### DISCUSSION TITLE
Optimal strategy for hybrid (Illumina + ONT) assembly

### DISCUSSION THREAD

**Discussion by MaestSi** [ablab/spades#829] [Q&A] [answered]
Dear all,
I am about to use rnaSPAdes v3.15.3 to perform hybrid transcriptome assembly, using both Illumina and ONT data.
Would you advise performing preprocessing of ONT reads obtained with PCR cDNA sequencing with tools as [pychopper](https://github.com/nanoporetech/pychopper), in order to identify, orient, trim and rescue full cDNA reads?
In case the answer is yes, would it be better to provide rnaSPAdes with ONT reads using --fl-rna instead of --nanopore parameter?
Thanks in advance,
Simone

**Comment by andrewprzh** [comment, marked as answer]
Hi!

We have not tried this tool in combination with rnaSPAdes assembly. However, it seems like you may give it a try.
And yes, you are absolutely right, detected FL reads are better be provided with --fl-rna option.

If you have time and a way to evaluate quality of your assembly, I'd suggest to try different pipelines:
- give rnaSPAdes raw ONT reads (--nanopore)
- give rnaSPAdes FL cDNA reads obtained with pychopper (--fl-rna)
- give both sets of reads using respective options.

Best
Andrey

**Reply by MaestSi** [reply]
Thank you for the prompt information, I'll probably give all the pipelines a try.
Best,
Simone

**Reply by MaestSi** [reply]
Hi, just an update on the tests I have been performing. Since pychopper identified almost 85% of reads as passing the quality check (both primers found, quality > 7, length > 50bp), I only used those reads, and I compared --fl-rna and --nanopore options. Based on BUSCO analysis, I obtained better results using --fl-rna option.
Best,
Simone

**Reply by andrewprzh** [reply]
Thank you for the information! Hope this thread may be of use for others too!
Best
Andrey
